### PR TITLE
Port `ae` commands to RzShell

### LINKED ISSUE #1533: Migrate `ae` commands to the rzshell
To make ESIL API more consistent and ease the migration from ESIL to RZIL.

See:
- `librz/core/cmd_analysis.c`
- `librz/core/cmd_descs/cmd_analysis.yaml`

### Todo

**Remove**

- [ ] `ae*` - https://github.com/rizinorg/rizin/pull/4730

**To RzShell, but remove with ESIL**
- [ ] `aepc`
- [ ] `ae `
- [ ] `aex`

**To RzShell, but port to RzIL**

- [ ] `aeb`
- [ ] `aef`
- [ ] `aefa`
- [ ] `aeA`
- [ ] `aea`
- [ ] `aeab`
- [ ] `aeafj`
- [ ] `aeafj`
- [ ] `aeab`
- [ ] `aeabj`

**To RzShell, but reimplement**
- [ ] `aek`
- [ ] `aek`
- [ ] `aek `
- [ ] `aek-`
- [ ] `aeli`
- [ ] `aeli`
- [ ] `aelir`

### PR DESCRIPTION
**Pull request by Rot127**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [x] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Port the `ae` commands to RzShell.

**Remove**

- [x] `ae*` - It was marked as not working and had no direct advantage. With the transition to RzIL it becomes obsolete.

**To RzShell, but remove with ESIL**
- [x] `aepc`
- [x] `ae `
- [x] `ae??`
- [x] `aex`

**To RzShell, but port to RzIL**

- [x] `aeb`
- [x] `aef`
- [x] `aefa`
- [x] `aeA`
- [x] `aea`
- [x] `aeab`
- [x] `aeafj`
- [x] `aeafj`
- [x] `aeab`
- [x] `aeabj`

**To RzShell, but reimplement**
- [x] `aek`
- [x] `aek-`
- [x] `aeli`
- [x] `aeli`
- [x] `aelir`

**Last**

- [x] Remove `a*` (doesn't work)
- [x] Remove legacy `rz_cmd_analysis` handler + fixup in yaml files.

**Test plan**

All green

**Closing issues**

closes https://github.com/rizinorg/rizin/issues/1533

### CODE REVIEW

File: librz/core/cmd_descs/cmd_analysis.yaml [hunk 1]
```diff
@@ -2402,3 +2402,139 @@ commands:
...
+        args:
+          - name: expr
+            type: RZ_CMD_ARG_TYPE_STRING
+            optional: true
+      - name: aeH
+        summary: Show ESIL help.
+        cname: analyze_esil_expr_help
+        args: []
+      - name: aeb
+        summary: Emulate current block.
```
Comment (XVilka) [inline librz/core/cmd_descs/cmd_analysis.yaml]: "Emulate current block using ESIL VM" or something like that. Until ESIL completely removed we should make distinction clear.

File: librz/core/cmd_descs/cmd_analysis.yaml [hunk 2]
```diff
@@ -2402,3 +2402,139 @@ commands:
...
+        cname: analyze_esil_emulate_block
+        args: []
+      - name: aepc
+        summary: Set ESIL PC to given address.
+        cname: analyze_esil_set_pc
+        args:
+          - name: addr
+            type: RZ_CMD_ARG_TYPE_NUM
+      - name: aek
+        summary: SDB queries on ESIL.info.
```
Comment (XVilka) [inline librz/core/cmd_descs/cmd_analysis.yaml]: Unnecessary dot?

File: librz/core/cmd_descs/cmd_analysis.yaml [hunk 3]
```diff
@@ -2402,3 +2402,139 @@ commands:
...
+        subcommands:
+          - name: aek
+            summary: Perform sdb query on ESIL.info.
```
Comment (XVilka) [inline librz/core/cmd_descs/cmd_analysis.yaml]: Same

File: librz/core/cmd_descs/cmd_analysis.yaml [hunk 4]
```diff
@@ -2402,3 +2402,139 @@ commands:
...
+            cname: analyze_esil_sdb_query
+            args:
+              - name: query
+                type: RZ_CMD_ARG_TYPE_STRING
+                default_value: "123*"
+          - name: aek-
+            summary: Resets the ESIL.info sdb instance.
```
Comment (XVilka) [inline librz/core/cmd_descs/cmd_analysis.yaml]: Same.

File: librz/core/cmd_descs/cmd_analysis.yaml [hunk 5]
```diff
@@ -2402,3 +2402,139 @@ commands:
...
+            cname: analyze_esil_sdb_reset
+            args: []
+      - name: aex
+        summary: Emulate instruction encoded in the given bytes.
```
Comment (XVilka) [inline librz/core/cmd_descs/cmd_analysis.yaml]: Same for all those emulation via ESIL instructions

File: librz/core/cmd/cmd_analysis.c [hunk 6]
```diff
@@ -6563,3 +6426,192 @@ RZ_IPI RzCmdStatus rz_analysis_data_trampoline_handler(RzCore *core, int argc, c
...
+
+RZ_IPI RzCmdStatus rz_analyze_esil_insn_access_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	int aea_mode = ((mode == RZ_OUTPUT_MODE_JSON) ? (1 << 4) : 0);
+	int len = rz_num_math(core->num, argv[2]);
+	bool is_aeA = argc == 4;
+	if (is_aeA) {
+		aea_mode |= 1;
+	}
+
+	// Don't ask me why this has to be done. I am just the janitor.
```
Comment (XVilka) [inline librz/core/cmd/cmd_analysis.c:6392]: Lol

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/core/cmd/cmd_analysis.c (hunk 6 above)
Comment (Rot127) [reply librz/core/cmd/cmd_analysis.c:6392]: It very much feels this way.

**Review by XVilka — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Pull request merged**